File: herencia-dto/SergioDeIscar-herencia-java/SergioDeIscar-herencia-java/src/main/java/org/example/models/Profesor.java
package org.example.models;

import org.example.dto.PersonaDto;

public class Profesor extends Persona{
    private final String modulo;

    public Profesor(String nombre, String modulo) {
        super(nombre);
        this.modulo = modulo;
    }

    public Profesor(Integer id, String nombre, String modulo) {
        super(id, nombre);
        this.modulo = modulo;
    }

    @Override
    public String toString() {
        return "Profesor{" +
                "id=" + getId() +
                ", nombre='" + getNombre() + '\'' +
                ", modulo='" + modulo + '\'' +
                '}';
    }

    public String getModulo() {
        return modulo;
    }

    @Override
    public PersonaDto toDto() {
        return new PersonaDto(Integer.toString(getId()), getNombre(), "Profesor", "null", getModulo());
    }
}

File: herencia-dto/Herencia_OscarGrC_Kotlin/src/main/kotlin/models/Profesor.kt
package models

import org.simpleframework.xml.Element
import org.simpleframework.xml.Root
import java.io.Serializable

data class Profesor (
    var modulo:String,
    override var nombre:String): Persona(),Serializable {
    override fun toString(): String {
        return "Hola soy ${nombre}, soy Profesor de $modulo \n"
    }
}